### DISCUSSION TITLE
Custom backend

### DISCUSSION THREAD

**Discussion by furesoft** [AvaloniaUI/Avalonia#9335] [Q&A] [answered]
how can i create a custom backend that replaced xaml with custom syntax?

**Comment by kekekeks** [comment, marked as answer]
You need to create a parser for your custom markup and MSBuild task to compile it.

Building object graphs is pretty straightforward, you can take a look at XAML compiler output by decompiling your view/user control in ILSpy.

**Comment by timunie** [comment]
Converted this into Q&A as this is neither a bug nor a feature request. If you disagree please let us know.
